manifests/snippets/secret_file.pp

```diff
@@ -13,7 +13,7 @@
     default  => $path
   }
   $safe_key = regsubst($hiera_key, '[^0-9A-Za-z_]', '_', 'G')
-  $data = hiera($hiera_key, hiera($safe_key))
+  $data = lookup($hiera_key, undef, undef, lookup($safe_key, undef, undef, undef))
```

Comment:
Inception?

Reply:
You think it should be brokendown into two lines instead?

Reply:
Two lines or maybe two variables but that might be what you ment? Easier to read and understand.